lib/iris/tests/unit/fileformats/grib/test_GribWrapper.py

```diff
@@ -60,97 +64,81 @@ def _mock_grib_get_native_type(grib_message, key):
     return result
 
 
-class Test_deferred(tests.IrisTest):
+class Test_edition(tests.IrisTest):
     def setUp(self):
-        confirm_patch = mock.patch(
-            'iris.fileformats.grib.GribWrapper._confirm_in_scope')
-        compute_patch = mock.patch(
-            'iris.fileformats.grib.GribWrapper._compute_extra_keys')
-        long_patch = mock.patch('gribapi.grib_get_long', _mock_grib_get_long)
-        string_patch = mock.patch('gribapi.grib_get_string',
-                                  _mock_grib_get_string)
-        native_patch = mock.patch('gribapi.grib_get_native_type',
-                                  _mock_grib_get_native_type)
-        confirm_patch.start()
-        compute_patch.start()
-        long_patch.start()
-        string_patch.start()
-        native_patch.start()
-        self.addCleanup(confirm_patch.stop)
-        self.addCleanup(compute_patch.stop)
-        self.addCleanup(long_patch.stop)
-        self.addCleanup(string_patch.stop)
-        self.addCleanup(native_patch.stop)
-
-    def test_regular_sequential(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
-        auto_regularise = False
+        self.patch('iris.fileformats.grib.GribWrapper._confirm_in_scope')
+        self.patch('iris.fileformats.grib.GribWrapper._compute_extra_keys')
+        self.patch('gribapi.grib_get_long', _mock_grib_get_long)
+        self.patch('gribapi.grib_get_string', _mock_grib_get_string)
+        self.patch('gribapi.grib_get_native_type', _mock_grib_get_native_type)
+        self.tell = mock.Mock(side_effect=[_message_length])
+
+    def test_not_edition_1(self):
+        def func(grib_message, key):
+            return 2
+
+        emsg = "GRIB edition 2 is not supported by 'GribWrapper'"
+        with mock.patch('gribapi.grib_get_long', func):
+            with self.assertRaisesRegexp(TranslationError, emsg):
+                GribWrapper(None)
+
+    def test_edition_1(self):
         grib_message = 'regular_ll'
-        for i, _ in enumerate(tell_tale):
-            gw = GribWrapper(grib_message, grib_fh, auto_regularise)
-            self.assertIsInstance(gw._data, NumpyArrayAdapter)
-            proxy = gw._data.concrete
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (10, 20))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, _message_length * i)
-            self.assertEqual(proxy.regularise, auto_regularise)
-
-    def test_regular_mixed(self):
```

Comment:
Are there going to be lots of merge conflicts in this file because of the stuff that Laura and I have been doing this week on her branch?

I am a bit reluctant to pass comment on this file as it might just be irrelevant...